File: queue.cpp
#include <iostream>
#define size 5 //Define the size of your queue

using namespace std;

class queue {
	int data[size];
	int tail;

public:

	queue() {
		tail = 0; //So this shows the next empty position
	}

	void insert() {

		if (tail == size) {
			cout<<"\nThe queue is full. Consider adding some elements.";
		}

		else {
			cout<<"\nData: ";
			cin>>data[tail++];
		}
	}

	void delete_data() {

		if (tail == 0) {
			cout<<"\nThe queue is empty. There is nothing to delete.";
		}

		else {
			cout<<"\nRemoved data: "<<data[0];
			for(int i = 1; i < tail; i++) {
				data[i - 1] = data[i];
			}

			tail --;
		}
	}

	void display() {
		cout<<"Queue: ";
		for(int i = 0; i < tail; i++) {
			if (i == (tail - 1)) {
				cout<<data[i];
			}

			else {
				cout<<data[i]<<" -> ";
			}
		}
	}
};

int main() {
	queue q;
	bool again = true;

	while(again) {
		int c;
		cout<<"\nOptions:\n1. Insert\n2. Delete\n3. Display\n4. Exit\n";
		cin>>c;

		switch(c) {
			case 1:	q.insert();
					break;

			case 2: q.delete_data();
					break;

			case 3: q.display();
					break;

			case 4: again = false;
					break;

			default: cout<<"\nWrong option. Try again";
		}
	}

	return 0;
}

File: queue.py
#!/usr/bin/env python3.4

class queue:
    def __init__(self):
        """Initialize stuff for the queue"""
        self.data = []

    def enqueue(self):
        """Insert data into the stack"""
        data = int(input("\nEnter the data: "))
        self.data.append(data)

    def dequeue(self):
        """Delete data from the stack"""
        if len(self.data) == 0:
            print("\nThere is nothing in the stack.")

        else:
            print("\nDeleted element is : ", self.data[0])
            self.data.pop(0)

    def display(self):
        """Display the elements in the stack"""
        print("\nQueue : ")
        
        x = len(self.data)

        for i in range(x):

            if i == (x - 1):
                print(self.data[i])

            else:
                print(self.data[i], end = " - ")

q = queue()

again = True

while(again):
    a = int(input("\nOptions \n1. Enqueue\n2. Dequeue\n3. Display\n4. Exit\n  "))

    if a == 1:
        q.enqueue()
    elif a == 2:
        q.dequeue()
    elif a == 3:
        q.display()
    elif a == 4:
        again = False
    else:
        print("\nWrong option selected. Try again.")
